sherpa/models/model.py

```diff
@@ -572,21 +578,25 @@ def regrid(self, *arrays, **kwargs):
         >>> request_space = np.arange(1, 10, 0.1)
         >>> regrid_model = mybox.regrid(request_space, interp=linear_interp)
         """
-        valid_keys = ('interp', )
-        for key in kwargs.keys():
-            if key not in valid_keys:
-                raise TypeError("unknown keyword argument: '%s'" % key)
-        eval_space = EvaluationSpace1D(*arrays)
+        try:
+            self.check_regrid_kwargs(**kwargs)
+        except TypeError as te:
+            raise te
```

Comment:
You don't need the `try`/`except` case here as it doesn't add anything (the exact same exception is re-raised).